mm2src/mm2_bitcoin/chain/src/block_header.rs

```diff
@@ -242,7 +241,7 @@ impl Deserializable for BlockHeader {
         };
         let nonce = if is_zcash {
             BlockHeaderNonce::H256(reader.read()?)
-        } else if (version == KAWPOW_VERSION && !reader.coin_variant().is_btc())
+        } else if (version == KAWPOW_VERSION && reader.coin_variant().is_rvn())
```

Comment:
this went from very broad (`!btc`) to very specific (`==rvn`). how does a coin like CHTA match this case?
i see we already have a test for CHTA so it gotta work, but wouldn't this break for some other coins?

Reply:
`!reader.coin_variant().is_btc()` was added very broad wrongly in the past, it was while testing complete headers download and deserialization for BTC SPV for lightning security https://github.com/KomodoPlatform/komodo-defi-framework/pull/1452 but it turns out only rvn (that I am currently aware of) that uses `KAWPOW_VERSION` this way, and when CHTA used it, it deserialized in a wrong way as it's not BTC. About this breaking other coins, it might, but we can only rely on the test vectors / cases and if we face cases that break because of that, fix them and add them to the test cases until we have enough coverage. The alternative is to look at the codebase of each utxo coin we support and see if they use a standard header nonce with `KAWPOW_VERSION` or not, I chose not to do that 😅